src/OrchardCore.Modules/OrchardCore.ContentFields/Extensions/ContentPartFieldDefinitionExtensions.cs

```diff
@@ -13,7 +13,7 @@ public static class ContentPartFieldDefinitionExtensions
         ContentItem contentItem)
         where TField : ContentField
     {
-        if (((JsonObject)contentItem.Content)[fieldDefinition.PartDefinition.Name] is not JsonObject jPart ||
+        if (((JsonObject)contentItem.Content)[fieldDefinition.ContentTypePartDefinition.Name] is not JsonObject jPart ||
```

Comment:
For those wondering: `fieldDefinition.PartDefinition.Name` here will always be the technical name of the content part itself, as also listed under /Admin/ContentTypes/ListParts. `fieldDefinition.ContentTypePartDefinition.Name` will be the same for non-reusable parts, however, it'll be the instance name for reusable named parts.